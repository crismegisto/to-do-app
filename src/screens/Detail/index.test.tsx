import React from 'react';
import { fireEvent, screen, waitFor } from '@testing-library/react-native';

import { initialStoreState, renderWithProvider } from '../../../utils/test/redux';
import TaskDetail from './index';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn()
    }),
    useRoute: () => ({ params: { id: 1 } })
  };
});

beforeEach(() => {
  renderWithProvider(<TaskDetail />, {
    preloadedState: initialStoreState
  });
});

describe('testing Detail', () => {
  test('snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('edit task', () => {
    const button = screen.getByText('Save');
    expect(button).toBeOnTheScreen();
  });
});
