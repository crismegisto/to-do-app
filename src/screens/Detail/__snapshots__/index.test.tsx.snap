// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Detail snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "margin": 14,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Name
  </Text>
  <TextInput
    accessibilityLabel="name input"
    maxLength={30}
    onChangeText={[Function]}
    style={
      {
        "borderWidth": 1,
        "height": 40,
        "marginVertical": 12,
        "padding": 10,
      }
    }
    value="Mock"
  />
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Description (optional)
  </Text>
  <TextInput
    maxLength={120}
    multiline={true}
    onChangeText={[Function]}
    style={
      {
        "borderWidth": 1,
        "height": 100,
        "marginVertical": 12,
        "padding": 10,
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 10,
        "opacity": 1,
        "padding": 10,
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Save
    </Text>
  </View>
</RCTSafeAreaView>
`;
